File: stack.java
import java.util.Scanner;

public class stack {
    int arr[];
    int top;
    
    stack()  // constructor
    {
            
        arr = new int[5];
        top = -1;
    
    }
    
    void push() // insert
    {
        if (top==4) 
        {
            System.out.println("Stack is Overflow...");
        }
        else
        {
            System.out.print("Enter the element: ");
            Scanner sc2 = new Scanner(System.in);
            int data = sc2.nextInt();
            top = top + 1;
            arr[top] = data;
            System.out.println("Data Instered..");
        }
    }

    void pop() // Delete
    {
        if (top==-1) {
            System.out.println("Stack is Underflow...");
        }
        else
        {
            System.out.println("Deleted Element: "+arr[top]);
            top = top - 1; 
        }

    }

    void traverse() // Show
    {
        if(top==-1)
        {
            System.out.println("Stack is Empty...");
        }
        else{
            for (int i = top; i >=0; i--) {
                System.out.println(arr[i]);
            }
        }
    }


    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        stack obj = new stack();
        while (true) {

            System.out.println("1- PUSH");
            System.out.println("2- POP");
            System.out.println("3- SHOW");
            System.out.println("4- EXIT");
            System.out.print("Enter Your Choice (1-4): ");
    
            int option = sc.nextInt();
            if (option == 1) {
                obj.push();
            } 
            else if (option == 2) {
                obj.pop();
            } 
            else if (option == 3) {
                obj.traverse();
            } 
            else if (option == 4) {
                break;
            } 
            else {
                
                System.out.println("Wrong Choice, Try Again !!");
            }
            System.out.println("\n----------------------------------\n");
        }
    }
}

File: stack.py
# Stack Operation
#  1 - Push 
#  2 - Pop
#  3 - Display
#  4 - Peek     (Top Element of the Stack)



# .................STARTING.......................
# ................................................



stack = []                  # List of Stack
top = -1                    # or None 


def push(stack):
    global top
    item = int(input("Enter the Element: "))
    stack.append(item)
    top = top + 1 

def pop(stack):
    global top
    if stack == []:
        print("Stack is Empty...... (Or Underflow)")
    else:
        print("Deleted Element:",stack.pop())
        top = top - 1

def peek(stack):
    if stack == []:
        print("Stack is Empty.")
    else:
        print("Top of the Stack:",stack[top])
    

def display(stack):
    if stack == []:
        print("Stack is Empty.")
    else:
        print("\n\n......Stack.......\n")  # Ignore
        for i in range(top,-1,-1):
            print(stack[i])
        print("\n..................\n\n")  # Ignore

# Starting of the Program..

while(True):
    print(".......STACK OPERATIONS........")
    print("1 - PUSH")
    print("2 - POP")
    print("3 - DISPLAY")
    print("4 - PEEK (Top Element of the Stack)")
    print("5 - Exit from the System")
    ch = int(input("Enter Your Choice (1-5): "))

    if ch == 1:
        push(stack)
    elif ch == 2:
        pop(stack)
    elif ch == 3:
        display(stack)
    elif ch == 4:
        peek(stack)
    elif ch == 5:
        exit(0)      # We can also use Break
    else:
        print("Invaild Choice")
